File: src/Implicit/ImplicitPow.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitPow : ImplicitModuleBase
    {
        public ImplicitPow(ImplicitModuleBase source, double power)
        {
            this.Source = source;
            this.Power = new ImplicitConstant(power);
        }

        public ImplicitPow(ImplicitModuleBase source, ImplicitModuleBase power)
        {
            this.Source = source;
            this.Power = power;
        }

        public ImplicitModuleBase Source { get; set; }

        public ImplicitModuleBase Power { get; set; }

        public override double Get(double x, double y)
        {
            return Math.Pow(this.Source.Get(x, y), this.Power.Get(x, y));
        }

        public override double Get(double x, double y, double z)
        {
            return Math.Pow(this.Source.Get(x, y, z), this.Power.Get(x, y, z));
        }

        public override double Get(double x, double y, double z, double w)
        {
            return Math.Pow(this.Source.Get(x, y, z, w), this.Power.Get(x, y, z, w));
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return Math.Pow(this.Source.Get(x, y, z, w, u, v), this.Power.Get(x, y, z, w, u, v));
        }
    }
}

File: src/implicit/ImplicitPow.java
package implicit;

public final class ImplicitPow extends ImplicitModuleBase {
    public ImplicitPow(final ImplicitModuleBase source, final double power) {
        this.Source = source;
        this.Power = new ImplicitConstant(power);
    }

    public ImplicitPow(final ImplicitModuleBase source, final ImplicitModuleBase power) {
        this.Source = source;
        this.Power = power;
    }

    private ImplicitModuleBase Source;

    private ImplicitModuleBase Power;

    public ImplicitModuleBase getSource() {
        return this.Source;
    }

    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }

    public ImplicitModuleBase getPower() {
        return this.Power;
    }

    public void setPower(final ImplicitModuleBase power) {
        this.Power = power;
    }

    @Override
    public double get(final double x, final double y) {
        return Math.pow(this.Source.get(x, y), this.Power.get(x, y));
    }

    @Override
    public double get(final double x, final double y, final double z) {
        return Math.pow(this.Source.get(x, y, z), this.Power.get(x, y, z));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w) {
        return Math.pow(this.Source.get(x, y, z, w), this.Power.get(x, y, z, w));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return Math.pow(this.Source.get(x, y, z, w, u, v), this.Power.get(x, y, z, w, u, v));
    }
}
